Kuroba/app/src/main/java/com/github/k1rakishou/chan/core/site/sites/wired7/Wired7Api.kt

```diff
@@ -0,0 +1,351 @@
+        "capcode" -> builder.moderatorCapcode(reader.nextString())
+        "extra_files" -> {
+          reader.beginArray()
+          while (reader.hasNext()) {
+            val postImage = readPostImage(reader, builder, board, endpoints)
+            if (postImage != null) {
+              files.add(postImage)
+            }
+          }
+          reader.endArray()
+        }
+        "md5" -> fileHash = reader.nextString()
+        "tn" -> fileThumb = reader.nextString()
+        else -> {
+          // Unknown/ignored key
+          reader.skipValue()
+        }
+      }
+    }
+
+    reader.endObject()
+
+    if (!builder.hasPostDescriptor()) {
+      Logger.e(TAG, "readPostObject() Post has no PostDescriptor!")
+      return
+    }
+
+    // The file from between the other values.
+    if (!fileId.isNullOrEmpty() && !fileExt.isNullOrEmpty()) {
+      val args = SiteEndpoints.makeArgument("tim", fileId, "ext", fileExt, "fpath", fpath.toString(), "tn", fileThumb)
+
+      val image = ChanPostImageBuilder()
+        .serverFilename(fileId)
+        .thumbnailUrl(endpoints.thumbnailUrl(builder.boardDescriptor, false, board.customSpoilers, args))
+        .spoilerThumbnailUrl(endpoints.thumbnailUrl(builder.boardDescriptor, true, board.customSpoilers, args))
+        .imageUrl(endpoints.imageUrl(builder.boardDescriptor, args))
+        .filename(Parser.unescapeEntities(fileName, false))
+        .extension(fileExt)
+        .imageWidth(fileWidth)
+        .imageHeight(fileHeight)
+        .spoiler(fileSpoiler)
+        .imageSize(fileSize)
+        .fileHash(fileHash, true)
+        .build()
+
+      // Insert it at the beginning.
+      files.add(0, image)
+    }
+
+    builder.postImages(files, builder.postDescriptor)
+
+    if (builder.op) {
+      // Update OP fields later on the main thread
+      val op = ChanPostBuilder()
+      op.closed(builder.closed)
+      op.archived(builder.archived)
+      op.sticky(builder.sticky)
+      op.replies(builder.totalRepliesCount)
+      op.threadImagesCount(builder.threadImagesCount)
+      op.uniqueIps(builder.uniqueIps)
+      op.lastModified(builder.lastModified)
+      chanReaderProcessor.setOp(op)
+    }
+
+    if (countryCode != null && countryName != null) {
+      val countryUrl = endpoints.icon("country", SiteEndpoints.makeArgument("country_code", countryCode))
+      builder.addHttpIcon(ChanPostHttpIcon(countryUrl, "$countryName/$countryCode"))
+    }
+
+    if (trollCountryCode != null && countryName != null) {
+      val countryUrl = endpoints.icon("troll_country", SiteEndpoints.makeArgument("troll_country_code", trollCountryCode))
+      builder.addHttpIcon(ChanPostHttpIcon(countryUrl, "$countryName/t_$trollCountryCode"))
+    }
+
+    chanReaderProcessor.addPost(builder)
+  }
+
+  @Throws(IOException::class)
+  private fun readPostImage(
+    reader: JsonReader,
+    builder: ChanPostBuilder,
+    board: ChanBoard,
+    endpoints: SiteEndpoints
+  ): ChanPostImage? {
+    try {
```

Comment:
I think you don't need this try/catch block since this is a workaround for 8kun bug.

Reply:
Fixed in [b4bfc0c](https://github.com/K1rakishou/Kuroba-Experimental/pull/750/commits/b4bfc0ca420ca71230f00819c457aa6c6b177461)